hammer/synthesis/genus/__init__.py

```diff
@@ -366,13 +375,29 @@ def syn_map(self) -> bool:
 
         return True
     
-    def ispatial_opt(self) -> bool:
-        if self.get_setting("synthesis.genus.phys_flow_effort").lower() == "high":
-            self.verbose_append("syn_opt -spatial")
-
+    def predict_floorplan(self) -> bool:
+        if self.get_setting("synthesis.genus.phys_flow_effort").lower() == "none":
+            self.verbose_append("set_db predict_floorplan_enable_during_generic false")
+            self.verbose_append("set_db physical_force_predict_floorplan false")
+        else:
+            self.verbose_append("set_db invs_temp_dir temp_invs")
+            self.verbose_append(f"set_db innovus_executable {self.get_setting('par.innovus.innovus_bin')}")
+            self.verbose_append("set_db predict_floorplan_enable_during_generic true")
+            self.verbose_append("set_db physical_force_predict_floorplan true")
+            self.verbose_append(f"set_db predict_floorplan_use_innovus true")
+            self.verbose_append("predict_floorplan")
         return True
 
     def add_tieoffs(self) -> bool:
+        
+        qrc_files = self.technology.read_libs([
+            hammer_tech.filters.qrc_tech_filter
+        ], hammer_tech.HammerTechnologyUtils.to_plain_item)
+
+        # using asap7 don't add tieoffs
```

Comment:
I don't recall why you needed to add this?

Reply:
Right after these lines I check the existence of qrc tech files to see if i run tieoffs or not. Do I not need to rerun it after it was executed in "init_environment"?

Reply:
Sorry, I meant why are tieoffs skipped for technologies that have QRC techfiles? Is this just due to the tieoffs bugs in 22.1?

Reply:
Yes, its due to the qrc bugs. The lines after skip tieoffs.

Reply:
Ok, then let's remove it before merging.

Reply:
done!